src/content/guides/packages.md

```diff
@@ -261,6 +261,174 @@ To update `pubspec.lock` run `dart pub get` without `--enforce-lockfile`.
 
 [content hash]: /tools/pub/glossary#content-hashes
 
+<a name="workspaces"></a>
+## Pub Workspaces (Mono-repo Support)
+
+When working on a project, you might develop multiple Dart packages in the same
+version control repository (a mono-repo).
+
+For example you might have a directory layout like: 
+
+```console
+/
+  packages/
+    shared/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+    client_package/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+    server_package/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+```
+
+There are multiple issues with this organization:
+
+* You need to run `pub get` once for each package.
+* You risk ending up with different versions of dependencies for each package.
+  Leading to confusion when context switching between the packages.
+* If you open the root folder in your IDE, the dart analyzer will have to have a
+  separate analysis contexts for each package.
+
+Pub allows you to organize your repository as a "workspace" using a single
+shared resolution for all your packages.
+
+To create a workspace:
+
+* Add a `pubspec.yaml` at the repository root directory with a `workspace` entry
+  enummerating the paths to the packages of the respository (the workspace
+  packages):
+
+```yaml
+name: _
+environment:
+  sdk: ^3.6.0
+workspace:
+  - packages/helper
+  - packages/client_package
+  - packages/server_package
+```
+
+* For each of the existing `pubspec.yaml`s, make sure their sdk constraint is at
+  least `^3.6.0` and add a `resolution` entry:
+
+  ```yaml
+  environment:
+    sdk: ^3.6.0
+  resolution: workspace
+  ```
+
+* Run `dart pub get` anywhere in the repository, and this will:
+  * Create a single `pubspec.lock` next to the root `pubspec.yaml` that contains
+    the a resolution of all the `dependencies` and `dev_dependencies` of all the
+    workspace packages. 
+  * Create a single shared `.dart_tool/package_config.json` that maps package
+    names to file locations.
+  * Delete any other existing `pubspec.lock` and
+    `.dart_tool/package_config.json` files next to workspace packages.
+
+Now the file structure looks like this:
+
+```console
+/
+  packages/
+    shared/
+      pubspec.yaml
+    client_package/
+      pubspec.yaml
+    server_package/
+      pubspec.yaml
+  pubspec.yaml
+  pubspec.lock
+  .dart_tool/package_config.json
+```
+
+:::version-note
+Pub workspaces was introduced in Dart 3.6.0.
+
+To use pub workspaces, all your workspace packages (but not your dependencies)
+must have a sdk version constraint of `^3.6.0` or higher.
+:::
+
+### Interdependencies between workspace packages
+
+If any of the workspace packages depend on each other, they will automatically
+resolve to the one in the workspace, regardless of the source.
+
+Eg. `packages/client_package/pubspec.yaml` might depend on `shared`:
+
+```yaml
+dependencies:
+  shared: 
+    hosted: https://pub.dev
```

Comment:
I understand why you want to write `hosted: https://pub.dev`

But we actually don't allow that to be published to pub.dev.

We require that packages published to pub.dev has no URL when they are hosted.
We make this requirement because we are the default repository.

Reply:
Good point.

Not sure how to fix this best. Changing it to a custom repository or a git-dependency seems distracting.

Reply:
Just writing `shared: ^1.0.0` might be okay.

Reply:
Done